logicrunner/executionbroker.go

```diff
@@ -343,6 +343,12 @@ func (q *ExecutionBroker) finishTask(ctx context.Context, transcript *Transcript
 }
 
 func (q *ExecutionBroker) processTranscript(ctx context.Context, transcript *Transcript) bool {
+	if transcript.Context != nil {
+		ctx = transcript.Context
+	} else {
+		inslogger.FromContext(ctx).Error("context in transcript is nil")
+	}
+
```

Comment:
Any chance we could fill ctx with context.Default() in tests instead?

Reply:
we decided to keep this check, howere in tests I made sure we don't have this error